app/controllers/jobseekers/profiles/professional_body_memberships_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Jobseekers::Profiles::ProfessionalBodyMembershipsController < Jobseekers::ProfilesController
+  helper_method :jobseeker_profile, :form, :professional_body_membership
+
+  def new; end
+  def edit; end
+
+  def create
+    if form.valid?
+      profile.professional_body_memberships.create!(professional_body_memberships_form_params)
+      redirect_to review_jobseekers_profile_professional_body_memberships_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    if form.valid?
+      professional_body_membership.update!(professional_body_memberships_form_params)
+      redirect_to review_jobseekers_profile_professional_body_memberships_path
+    else
+      render :edit
+    end
+  end
+
+  def confirm_destroy; end
+
+  def destroy
+    professional_body_membership.destroy!
+    redirect_to review_jobseekers_profile_professional_body_memberships_path, success: t(".success")
+  end
+
+  def form
+    @form ||= Jobseekers::ProfessionalBodyMembershipForm.new(form_attributes)
+  end
+
+  def form_attributes
```

Comment:
I have been trying to take this pattern out of the codebase. IMHO it's really hard to read, and collapses into simple code when taken out - @form = Form.new(params) is much easier to read?